src/ctr_logging.c

```diff
@@ -540,3 +565,108 @@ void sync_logs(void)
 		if (fsync(k8s_log_fd) < 0)
 			pwarn("Failed to sync log file before exit");
 }
+
+
+/* strlen("1997-03-25T13:20:42.999999999+01:00 stdout 9999999999 999999999 ") + 1 */
+#define TSSTREAMBUFLEN 128
+
+/*
+ * PROPOSED: CRI Stream Format, variable length file format
+ */
+static int set_k8s_stream_timestamp(char *buf, ssize_t bufsiz, ssize_t *tsbuflen, const char *pipename, uint64_t offset, ssize_t buflen, ssize_t *btbw)
```

Comment:
there seem to be some similarities between this and set_k8s_timestamp. can we either make the shared functionality a function or document why that's not possible?

Reply:
I did not want to disturb any existing code with this to avoid introducing un-intended bugs.

If we decide that it is worth merging this, then we should consider combing those two methods into one.